### ISSUE TITLE
electron-builder `info.plist` not work in Xcode13 & macOS 12.2

### ISSUE THREAD

**Issue by yujingxiacai** [electron-userland/electron-builder#6948]
- **Operator System**:   macOS Monterey 12.2.1
- **Xcode Version**:  13.3 (13E113)
- **Electron-Builder Version**:  ^22.14.13
- **Electron**:  ^15.1.0
- **Node Version**： v16.13.2

I have the config below as i need to access System Microphone，

>``` "mac": {
>	"target": [
>				"dmg"
>			],
>			"hardenedRuntime": true,
>			"entitlements": "assets/entitlements.mac.plist",
>			"entitlementsInherit": "assets/entitlements.mac.plist",
>			"gatekeeperAssess": false,
>			"category": "public.app-category.developer-tools",
>			"extendInfo": {
>      "NSMicrophoneUsageDescription":"This app requires microphone access to record audio."
>			}
>		},

### Phenomenon of my app
- install my app
  🎉 Normal...

- open my app 
   🎉Automatically recall MacOS `Security&Privacy` Setting
   Then I choose to allow my app to access the microphone, and reopen.

- reopen
   😭 **My `checkMacAudio` logic crash & white screen ⚠️**
- open app Contents
     Find the `info.plist`, and open it with Xcode, Press the shortcut Command + S to save it (No modification required）
-  reopen
    My `checkMacAudio` logic executes fine, app works fine 
-  😵
  Why? is a not yet adapted to Xcode13?

https://github.com/electron-userland/electron-builder/issues/6944#issuecomment-1157541157

**Comment by yujingxiacai** [comment]
Does anyone know why this is, I am not an apple developer, so I don't know much about this series of related issues

**Comment by bechurch** [comment]
Having this issue as well

**Comment by yujingxiacai** [comment]
**Excellent, my question have been resolved🎉🎉🎉 lol.**

**My solution is as follows:**
----

- First let's look at the `Crash reports` in the `Console` app

```
-------------------------------------
Translated Report (Full Report Below)
-------------------------------------

Process:               MyTest Helper (Renderer) [57980]
Path:                  /Applications/MyTest.app/Contents/Frameworks/MyTest Helper (Renderer).app/Contents/MacOS/MyTest Helper (Renderer)
Identifier:            com.SenseTime.MyTest.helper.Renderer
Version:               1.1.0
Code Type:             X86-64 (Native)
Parent Process:        MyTest [57959]
Responsible:           MyTest [57959]
User ID:               278765259

Date/Time:             2022-06-28 16:31:11.5874 +0800
OS Version:            macOS 12.2.1 (21D62)
Report Version:        12
Bridge OS Version:     6.4 (19P4242)
Anonymous UUID:        5B7G8E5E-A540-1242-B6B7-E8833713E9C6

Sleep/Wake UUID:       B875AB92-6WF9-4646-8BAC-535CB814CYEF

Time Awake Since Boot: 51000 seconds
Time Since Wake:       11626 seconds

System Integrity Protection: disabled

Crashed Thread:        1  Dispatch queue: com.apple.root.default-qos

Exception Type:        EXC_CRASH (SIGABRT)
Exception Codes:       0x0000000000000000, 0x0000000000000000
Exception Note:        EXC_CORPSE_NOTIFY

Termination Reason:    Namespace TCC, Code 0 
This app has crashed because it attempted to access privacy-sensitive data without a usage description. 
The app's Info.plist must contain an com.apple.security.device.audio-input key with a string value explaining 
to the user how the app uses this data.

Application Specific Information:
dyld4 config: DYLD_INSERT_LIBRARIES=/usr/local/lib/libesafenet.dylib
```

-----

-  Above Critical error description
     **This app has crashed because it attempted to access privacy-sensitive data without a usage description. The app's Info.plist must contain an com.apple.security.device.audio-input key with a string value explaining to the user how the app uses this data.**
I am sure have `"NSMicrophoneUsageDescription": "This app requires microphone access to record audio.",`  Why still tell i have not yet set this config item.
Then, i have saw this blog [Electron App 在 MacOS 下 申请摄像头及麦克风权限踩坑指南](https://www.bugs.cc/p/electron-app-request-camera-and-microphone-permission-by-macos/),


> my `entitlements` file content as below
```<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
  <dict>
    <key>com.apple.security.cs.allow-unsigned-executable-memory</key>
    <true/>
    <key>com.apple.security.cs.allow-jit</key>
    <true/>
    <key>com.apple.security.network.client</key>
    <true/>
    <key>com.apple.security.device.audio-input</key>
    <true/>
    <key>com.apple.security.cs.allow-dyld-environment-variables</key>
    <true/>
    <key>com.apple.security.cs.disable-library-validation</key>
    <true/>
  </dict>
</plist>
```

Yes! same as this blog...
Where is the problem?

- Could it be the reason why the signature and notarization were not done?
   Then I signed the app and notarized it. Unfortunately, the response i have got was `The Mac software that you uploaded was not notarized.`.
 I checked the notary error log find some of my code have not `Hardened Runtime`, and then i resolved this problem.
but my app still crash..
   
- It must be the problem that the system fails to authorize the audio
  Through the leave-one-out method, I found that it is definitely not an `"NSMicrophoneUsageDescription": "This app requires microphone access to record audio.",` configuration problem.

- Find the problem: `entitlements.mac.plist` not truly work
   **`entitlements.mac.plist` must be used in 'sign' period**
   For historical reasons, I skipped the sign of the `electron-builder`, but signed it through a custom `electron-osx-sign`.
So, package.json config of  entitlements not work.

I change my `electron-osx-sign` command line to `electron-osx-sign --hardened-runtime --no-gatekeeper-assess --entitlements=./assets/entitlements.mac.plist --entitlements-inherit=./assets/entitlements.mac.plist ./release/build/mac/MyTest.app --keychain=/Library/Keychains/System.keychain`

> old command line:
> `electron-osx-sign --hardened-runtime --no-gatekeeper-assess  ./release/build/mac/MyTest.app --keychain=/Library/Keychains/System.keychain`

Then I notarized the program and solved the crash problem...


To sum up, you need to use the entitlements file to open your audio permissions when signing.

**Comment by bechurch** [comment]
@yujingxiacai This worked. thank you!

**Comment by hrana123** [comment]
Step 1: Create the entitlement.plist file in Document and paste these keys.

com.apple.security.inherit com.apple.security.cs.allow-jit com.apple.security.cs.allow-unsigned-executable-memory com.apple.security.cs.disable-library-validation com.apple.security.cs.allow-dyld-environment-variables com.apple.security.cs.debugger
<key>com.apple.security.device.camera</key>
<true/>
com.apple.security.device.microphone
com.apple.security.network.server

com.apple.security.network.client

com.apple.security.device.usb

com.apple.security.print

com.apple.security.device.bluetooth

com.apple.security.files.user-selected.read-only
com.apple.security.files.user-selected.read-write

com.apple.security.files.downloads.read-only

com.apple.security.files.downloads.read-write

com.apple.security.assets.pictures.read-write

com.apple.security.assets.music.read-only
com.apple.security.assets.music.read-write

com.apple.security.assets.movies.read-only

com.apple.security.assets.movies.read-write
</dict>
Step 2: Right click in the application go to contents then go to info.plist file and add this key.
NSCameraUsageDescription
<Application_name> application requests access to the device's camera.

Step 3: Run this command in terminal .
sudo codesign --sign -
--entitlements /Users/devadmin/Documents/entitlements.plist
--deep /Applications/Talygen.app
--force

Step 4: After that run the application and access the camera.
